data/preservedata.rb

```diff
@@ -0,0 +1,171 @@
+require_relative '../book'
+require_relative '../person'
+require_relative '../student'
+require_relative '../teacher'
+require 'json'
+
+def save_book(book)
+  book_obj = {
+    title: book.title,
+    author: book.author
+  }
+
+  file = File.open('./data/book.json')
+
+  if file.read.empty?
+    book = [book_obj]
+  else
+    book = JSON.parse(File.read('./data/book.json'))
+    book << book_obj
+  end
+
+  file.close
+
+  myfile = File.open('./data/book.json', 'w')
+  myfile.write(JSON.pretty_generate(book))
+  myfile.close
+end
+
+def books_stored
+  file = File.open('./data/book.json')
+
+  if File.empty?('./data/book.json')
+    'Please add some books.  No books available.'
+  else
+    books = JSON.parse(File.read('./data/book.json'))
+    puts 'Books on our list:'
+    books.each do |book|
+      book = Book.new(title: book['title'], author: book['author'])
+      @books_array << book
+    end
+  end
+  file.close
+  @books_array.each do |book|
+    puts "Title: #{book.title}, Author: #{book.author}"
+  end
+end
+
+def save_student(student)
+  student_obj = {
+    id: student.id,
+    name: student.name,
+    age: student.age,
+    parent_permission: student.parent_permission,
+    classroom: student.classroom
+  }
+
+  file = File.open('./data/person.json')
+
+  if file.read.empty?
+    person = [student_obj]
+  else
+    person = JSON.parse(File.read('./data/person.json'))
+    person << student_obj
+  end
+
+  file.close
+
+  myfile = File.open('./data/person.json', 'w')
+  myfile.write(JSON.pretty_generate(person))
+  myfile.close
+end
+
+def save_teacher(teacher)
+  teacher_obj = {
+    id: teacher.id,
+    name: teacher.name,
+    age: teacher.age,
+    specialization: teacher.specialization
+  }
```

Comment:
![image](https://user-images.githubusercontent.com/93039359/225156360-eb02d357-6cd3-449e-b449-f73bb0d043c9.png)


- [ ] You are adding this object to the same file where students are so it would be nice if you add a key-value pair of `type: 'teacher'` to not get confused about who is the teacher and who is the student :+1: and also it will help you in your project as well :blush: